tools/clang/test/CodeGenDXIL/hlsl/objects/NodeObjects/mesh-node.hlsl

```diff
@@ -0,0 +1,130 @@
+// RUN: %dxc -T lib_6_9 %s | FileCheck %s
+
+// REQUIRES: dxil-1-9
+
+// Test loading of node input and funneling into mesh outputs
+// Essentially an end-to-end mesh node test.
+
+
+RWBuffer<float> buf0;
+
+#define MAX_VERT 32
+#define MAX_PRIM 16
+#define NUM_THREADS 32
+
+struct MeshPerVertex {
+    float4 position : SV_Position;
+    float color[4] : COLOR;
+};
+
+struct MeshPerPrimitive {
+    float normal : NORMAL;
+    float malnor : MALNOR;
+    float alnorm : ALNORM;
+    float ormaln : ORMALN;
+    int layer[6] : LAYER;
+};
+
+struct MeshPayload {
+    float normal;
+    float malnor;
+    float alnorm;
+    float ormaln;
+    int layer[6];
+};
+
+groupshared float gsMem[MAX_PRIM];
+
+[Shader("node")]
+[NodeLaunch("mesh")]
+[outputtopology("triangle")]
+[numthreads(128, 1, 1)]
+[NodeDispatchGrid(64,1,1)]
+void node_setmeshoutputcounts(DispatchNodeInputRecord<MeshPayload> mpl,
+            out indices uint3 primIndices[MAX_PRIM],
+            out vertices MeshPerVertex verts[MAX_VERT],
+            out primitives MeshPerPrimitive prims[MAX_PRIM],
+            in uint tig : SV_GroupIndex) {
+  // SV_GroupIndex
+  // CHECK: %[[GID:[0-9]+]] = call i32 @dx.op.flattenedThreadIdInGroup.i32(i32 96)
+
+  // CHECK: call void @dx.op.setMeshOutputCounts(i32 168, i32 32, i32 16)  ; SetMeshOutputCounts(numVertices,numPrimitives)
+  SetMeshOutputCounts(32, 16);
+
+  // create mpl
+  // CHECK: %[[CHDL:[0-9]+]] = call %dx.types.NodeRecordHandle @dx.op.createNodeInputRecordHandle(i32 250, i32 0)
+  // CHECK: %[[AHDL:[0-9]+]] = call %dx.types.NodeRecordHandle @dx.op.annotateNodeRecordHandle(i32 251, %dx.types.NodeRecordHandle %[[CHDL]], %dx.types.NodeRecordInfo { i32 97, i32 40 })
+
+  MeshPerVertex ov;
+  ov.position = float4(14.0,15.0,16.0,17.0);
+  ov.color[0] = 14.0;
+  ov.color[1] = 15.0;
+  ov.color[2] = 16.0;
+  ov.color[3] = 17.0;
+
+  if (tig % 3) {
```

Comment:
Not that we'd ever run this code, but it doesn't make much sense (should be `(!(tig % 3))`), and has data race problems (with `gsMem`).  I see this appears to be based on some other mesh shader test code copied all over the place, so whatever. 🤷

Basically, if we were to implement some useful diagnostics for obvious data races such as this, it would not be happy with this code.

Reply:
Note taken!